### PR TITLE
Add setup.py to make library installable via pip

### LINKED ISSUE #7: Make pip installable

### PR DESCRIPTION
**Pull request by mschmidt87**
This PR adds a setup script to the repository so that the library can be installed via `pip install .` .
This fixes #7 .

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# encoding: utf8
+from setuptools import setup
+
+setup(
+    name='python-gp',
+    version='0.1',
+    author='Jakob Jordan, Maximilian Schmidt',
+    author_email='j.jordan@fz-juelich.de',
```
Comment (jakobj) [inline setup.py]:
```suggestion
    author_email='jakobjordan@posteo.de',
```

File: setup.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    description=('Cartesian genetic programming in Python.'),
+    license='GPLv3',
+    keywords='genetic programming',
+    url='https://github.com/jakobj/python-gp',
+    python_requires='>=3.6, <4',
+    install_requires=['sympy', 'torch', 'numpy', ],
```
Comment (jakobj) [inline setup.py]: shouldn't we read this from requirements.txt?

File: setup.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
+# encoding: utf8
+from setuptools import setup
+
+setup(
+    name='python-gp',
+    version='0.1',
+    author='Jakob Jordan, Maximilian Schmidt',
+    author_email='j.jordan@fz-juelich.de',
+    description=('Cartesian genetic programming in Python.'),
```
Comment (jakobj) [inline setup.py]:
```suggestion
    description=('Cartesian Genetic Programming in Python.'),
```

File: setup.py (hunk 2 above)
Comment (mschmidt87) [reply setup.py]: requirements file is introduced in #23 , so let's merge #23 first and I'll modify here then.

**Pull request merged**